### PR TITLE
Add API Token management

### PR DESCRIPTION
**Pull request by thisisgoldman**
<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/ps7MKmCULkGcazvFyiEe/e1dff82c-d0f8-4517-912a-a7d1fb8e288c.mp4">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/ps7MKmCULkGcazvFyiEe/e1dff82c-d0f8-4517-912a-a7d1fb8e288c.mp4">CleanShot 2024-11-07 at 15.02.01.mp4</video>

## Add API Token Management

* Users can create and delete API tokens
* Users can toggle which API token is presented when given API and CLI instructions

### CODE REVIEW

**Comment by thisisgoldman** [comment]
* **#513** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/513?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @thisisgoldman and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: assets/svelte/components/CodeWithSecret.svelte [hunk 1]
```diff
@@ -48,28 +63,56 @@
...
+            <icon
+              class:hidden={!showApiToken}
+              class="hero-eye-slash text-gray-500"
+            />
+          </button>
+        </div>
+        <div class="cursor-pointer">
+          <CopyIcon content={codeForCopy} />
+        </div>
+      {:else}
```
Comment (thisisgoldman) [inline assets/svelte/components/CodeWithSecret.svelte:106]: If there are no API keys, then tell the user to go create an API key.

File: assets/svelte/components/CodeWithSecret.svelte [hunk 2]
```diff
@@ -48,28 +63,56 @@
...
-          />
-        </button>
-      </div>
-      <div class="cursor-pointer">
-        <CopyIcon content={codeForCopy} />
-      </div>
+      {#if apiTokens.length > 0}
+        <div class="flex flex-row items-center space-x-2">
+          <Label class="text-sm text-gray-500">API Token</Label>
+          <Select
```
Comment (thisisgoldman) [inline assets/svelte/components/CodeWithSecret.svelte:71]: Allow the user to select which API key is shown in the code snippets

File: lib/sequin/api_tokens/api_token.ex [hunk 3]
```diff
@@ -26,6 +26,7 @@ defmodule Sequin.ApiTokens.ApiToken do
     |> cast(params, [:name, :token])
     |> validate_required([:name, :token])
     |> Sequin.Changeset.validate_name()
+    |> unique_constraint([:name, :account_id], name: "api_tokens_account_id_name_index")
```
Comment (thisisgoldman) [inline lib/sequin/api_tokens/api_token.ex:29]: There is a unique index on the DB on account_id and name - so if the user tries to create an API token with the same name in the same account - everything explodes. This catches that.

File: lib/sequin_web/live/databases/form.ex [hunk 4]
```diff
@@ -448,6 +449,17 @@ defmodule SequinWeb.DatabasesLive.Form do
     }
   end
 
+  defp encode_api_tokens(api_tokens) when is_list(api_tokens) do
```
Comment (thisisgoldman) [inline lib/sequin_web/live/databases/form.ex:452]: I know I'm re-using this encode function in multiple places - but that seemed like the pattern we like. Keep encoding in context of the FE module where it's used

File: assets/svelte/components/CodeWithSecret.svelte [hunk 5]
```diff
@@ -48,28 +63,56 @@
...
+              class:hidden={!showApiToken}
+              class="hero-eye-slash text-gray-500"
+            />
+          </button>
+        </div>
+        <div class="cursor-pointer">
+          <CopyIcon content={codeForCopy} />
+        </div>
+      {:else}
+        <Button href="/settings/accounts" variant="link" size="sm">
```
Comment (acco) [inline assets/svelte/components/CodeWithSecret.svelte:107]: Wrap in `LinkPatchNavigate`, don't use `href` directly

File: assets/svelte/settings/AccountSettings.svelte [hunk 6]
```diff
@@ -124,6 +145,39 @@
...
 
+  function handleCreateApiToken(event: SubmitEvent) {
+    createApiTokenLoading = true;
+    createApiTokenError = null;
+
+    live.pushEventTo(
+      `#${parent}`,
+      "create_api_token",
+      {
+        accountId: selectedAccount.id,
```
Comment (acco) [inline assets/svelte/settings/AccountSettings.svelte:156]: don't we know their current account?

**Pull request merged**